File: Py/704.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        length = len(nums)-1
        low = 0
        high = length
        
        while low<=high:
            mid = low + (high-low)//2
            
            if target == nums[mid]:
                return mid
            elif target<nums[mid]:
                high = mid-1
            else:
                low = mid+1
                
        return -1

File: CPP/704.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        
        int high, low=0, mid;
        
        high = nums.size()-1;
        
        while(low<=high){
            mid = (high-low)/2 + low;
            
            if(target == nums[mid])
                return mid;
            
            else if(target>nums[mid])
                low = mid+1;
            
            else
                high = mid-1;
            
        }
        
        return (-1);
        
    }
};